OrbitGl/GlCanvas.h

```diff
@@ -7,12 +7,13 @@
 
 #include "GlUtils.h"
 #include "ImGuiOrbit.h"
+#include "OrbitGlAccessibility.h"
 #include "PickingManager.h"
 #include "TextRenderer.h"
 #include "TimeGraph.h"
 #include "Timer.h"
 
-class GlCanvas {
+class GlCanvas : public orbit_gl::GlAccessibleInterface {
```

Comment:
I'm not sure about inheritance here. I'm wondering if on the OrbitGl side we shouldn't stick to a very generic  bounding box hierarchy that could be translated at the OrbitQt level to the proper Qt accessibility object hierarchy, but could also be used at the OrbitGl level for rectangle selection in the capture view, or for changing the mouse cursor for example (b/174337266, b/174833325). In this scenario, the hierarchy would be populated by a call to a "GetBoundingBoxHierarchy()" method on the GlCanvas, that would in turn query the timegraph, which would query tracks, etc., removing the need for inheritance. Maybe I'm missing something, but it seems we could solve multiple issues at once!?

Reply:
Absolutely right. I think I should get rid of this inheritance. For all child controls (TimeGraph, Track...) I actually went a different route and created new classes inheriting from the interface that receive "their" control in the constructur and just query the relevant attributes. A control then owns an instance of the class and exposes it - so similar to the seperate tree that Qt is creating, but the nodes are actually owned by the controls themselves instead of tracked in a central registry.

Don't know why I didn't do the same for GlCanvas, I'll change that.

Reply:
I changed this now, PTAL. I'm actually thinking if we should name this "Interface", since it actually has implementation, if only for the constructor and destructor. Not sure if we have a strict definition for "Interface"?